core/storage/blog/gae_models.py

```diff
@@ -496,3 +507,144 @@ def create(cls, blog_post_id: str, author_id: str) -> BlogPostRightsModel:
         entity.put()
 
         return entity
+
+
+class BlogAuthorDetailsModel(base_models.BaseModel):
+    """Model for storing user's blog author details.
+
+    The id/key of instances of this model is randomly generated string of
+    length 12."""
+
+    # We use the model id as a key in the Takeout dict.
+    ID_IS_USED_AS_TAKEOUT_KEY: Literal[True] = True
+
+    author_id = datastore_services.StringProperty(indexed=True, required=True)
+    # The publicly viewable name of the user to display as author name in blog
+    # posts.
+    author_name = datastore_services.StringProperty(indexed=True)
```

Comment:
Maybe let's call this displayed_author_name to distinguish it from the author_id's username.

Is it not a required field?

Reply:
Done